core/fetcher.go

```diff
@@ -26,6 +26,21 @@ func NewBlockFetcher(client Client) *BlockFetcher {
 	}
 }
 
+// GetBlockInfo queries Core for additional block information, like Commit and ValidatorSet.
+func (f *BlockFetcher) GetBlockInfo(ctx context.Context, height *int64) (*types.Commit, *types.ValidatorSet, error) {
+	commit, err := f.Commit(ctx, height)
+	if err != nil {
+		return nil, nil, fmt.Errorf("core/fetcher: getting commit: %w", err)
```

Comment:
Can these errors be more descriptive? E.g. also include the `height`.

Reply:
Height will be logged out in the places using the code.

Reply:
That's doesn't surprise me. In general I feel like it would be better if errors were completely self-contained, and didn't depend on spooky logic on other places of the code to log out everything. Might just be a Go thing, so 🤷